### PR TITLE
psbt: Avoid unsigned int overflow in PSBT_IN_TAP_BIP32_DERIVATION

### LINKED ISSUE #25749: psbt.h:644:51: runtime error: unsigned integer overflow: 0 - 1 cannot be represented in type 'unsigned long'
Steps to reproduce:

* Build with `integer` sanitizer
* `UBSAN_OPTIONS="suppressions=$(pwd)/test/sanitizer_suppressions/ubsan:print_stacktrace=1:halt_on_error=1:report_error_type=1" ./src/qt/bitcoin-qt`
* Enter into the RPC debug console: `decodepsbt "cHNidP8BAKOro2MDAwMDA5ggCAAA////CQAtAAD+///1AAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJAAAAAAAAAAAAAAAAAAAAAAAAAD+///1Zm9ybmV3nWx1Y2vmelLmegAAAAAAAAAAAAAAAAAAAAMKAwMDAwMDAwMDAwMACvMBA3FkAAAAAAAAAAAABAAlAAAAAAAAACEWDQ0zDQ0NDQ0NDQ0NCwEAAH9/f39/fwMAAABNo6P///kAAA=="`

### PR DESCRIPTION
**Pull request by achow101**
Fixes #25749

### CODE REVIEW

**Review by darosior — APPROVED** [review]
utACK befa2a76da4adb5140b4870cd241c38e8ae2e0d0

**Comment by instagibbs** [comment]
can we catch this in test?

**Comment by achow101** [comment]
Added a test case.

**Comment by instagibbs** [comment]
ACK 70a55c059b014c7a687de7a4813a90c65148aed4

File: test/functional/rpc_psbt.py [hunk 1]
```diff
@@ -460,6 +461,9 @@ def run_test(self):
         # Invalid PSBTs
         for invalid in invalids:
             assert_raises_rpc_error(-22, "TX decode failed", self.nodes[0].decodepsbt, invalid)
+        for invalid in invalid_with_msgs:
+            psbt, msg = invalid
```
Comment (jonatack) [inline test/functional/rpc_psbt.py:465]: if you retouch
```diff
-        for invalid in invalid_with_msgs:
-            psbt, msg = invalid
+        for psbt, msg in invalid_with_msgs:
```

**Review by jonatack — COMMENTED** [review]
Review ACK 70a55c059b014c7a687de7a4813a90c65148aed4, this should avoid the issue reported in https://github.com/bitcoin/bitcoin/issues/25749

Error message with the new test, without the code change in `psbt.h`:

`TX decode failed Fee exceeds maximum configured by user (e.g. -maxtxfee, maxfeerate)`

Error message with the new test and the code change in `psbt.h`:

`TX decode failed Input Taproot BIP32 keypath has an invalid length`

Unrelated: it may be good as a follow-up to add a colon at the end of "TX decode failed" before appending the specific error message.

**Comment by darosior** [comment]
re-utACK 70a55c059b014c7a687de7a4813a90c65148aed4

**Pull request merged**